pstoremem/addr_book.go

```diff
@@ -26,31 +26,46 @@ func (e *expiringAddr) ExpiredBy(t time.Time) bool {
 	return t.After(e.Expires)
 }
 
-type addrSlice []expiringAddr
-
 var _ pstore.AddrBook = (*memoryAddrBook)(nil)
 
 // memoryAddrBook manages addresses.
 type memoryAddrBook struct {
 	addrmu sync.Mutex
-	addrs  map[peer.ID]addrSlice
+	addrs  map[peer.ID]map[string]expiringAddr
+
+	nextGC time.Time
 
 	subManager *AddrSubManager
 }
 
 func NewAddrBook() pstore.AddrBook {
 	return &memoryAddrBook{
-		addrs:      make(map[peer.ID]addrSlice),
+		addrs:      make(map[peer.ID]map[string]expiringAddr),
 		subManager: NewAddrSubManager(),
 	}
 }
 
+func (mab *memoryAddrBook) gc() {
```

Comment:
Could we add a comment to clarify that `gc()` is called within the context of an add/update, hence it piggybacks on their locking of the addr map?

Reply:
Done.